resources/idlharness.js

```diff
@@ -84,10 +84,26 @@ function minOverloadLength(overloads)
     .reduce(function(m, n) { return Math.min(m, n); });
 }
 
+// A helper to get the global of a Function object.  This is needed to determine
+// which global exceptions the function throws will come from.
+function globalOf(func)
+{
+    var global;
```

Comment:
Not sure if this works in JS, but could we just do:

```
try {
  return func.constructor("return this;")();
} catch (e) {}
return self;
```

Reply:
It depends on how defensive we're trying to be.  For example, that return statement inside the try/catch block could return `undefined` or something, depending on how off we are on what `func` is.  To be even more defensive, we could test that `global` is actually an `Object` and return `global.self || self`.  In practive, hopefully it all doesn't matter, so maybe we should just do the simple thing.